db/dbProvider.js

```diff
@@ -0,0 +1,12 @@
+class DBProvider {
+    constructor({ pool }) {
+        this.pool = pool;
+    }
+
+    async execute(query) {
+        const result = await this.pool.query(query);
+        return result.rows;
```

Comment:
I can be wrong, but I think `rows` could be `undefined` for the empty cases? Is it sustained?